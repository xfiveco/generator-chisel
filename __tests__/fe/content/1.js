const fs = require('fs-extra');
const { defaultAnswers, md, json, postPageSimple } = require('./helpers');

describe('FE Content', () => {
  test('Nested pages, alt templates, extra data', async () => {
    await global.chiselTestHelpers.generateProjectWithAnswers(
      ['create', '--skip-format-and-build'],
      defaultAnswers,
    );

    fs.outputFileSync('./content/nested/page.md', md('Hello Page'));
    fs.outputFileSync(
      './content/nested/more-nested/page.md',
      md('Hello More', '', 'date: 2020-04-27T12:30:12Z\nauthor: Jakub'),
    );
    fs.outputFileSync(
      './content/nested/more-nested/with-template.md',
      md(
        'Hello Alt',
        'This author writes things',
        'template: alternative\ndate: 2020-04-27T22:00:00Z\nauthor: Lubos',
      ),
    );
    fs.outputFileSync('./content/json-page.json', json({}));

    fs.outputFileSync(
      './content/nested/json-page.json',
      json({ template: 'alternative', title: 'JSON Title', author: 'Jason' }),
    );

    fs.outputFileSync(
      './src/templates/post.twig',
      postPageSimple(
        `
          {% if post.data.date %}
            Written on {{ post.data.date.toISOString() }}
            by {{ post.data.author }}
            in {{ post.data.category }}.
          {% endif %}
        `,
      ),
    );
    fs.outputFileSync(
      './src/templates/alternative.twig',
      postPageSimple(
        'Alternative template with post by {{ post.data.author }}',
      ),
    );

    await global.chiselTestHelpers.runChiselScript(['build']);

    await global.chiselTestHelpers.expectFilesToMatchSnapshot(['./dist']);
    await global.chiselTestHelpers.fileMatchesSnapshot(
      './dist/nested/page.html',
    );
    await global.chiselTestHelpers.fileMatchesSnapshot(
      './dist/nested/more-nested/page.html',
    );
    await global.chiselTestHelpers.fileMatchesSnapshot(
      './dist/nested/more-nested/with-template.html',
    );
    await global.chiselTestHelpers.fileMatchesSnapshot('./dist/json-page.html');
    await global.chiselTestHelpers.fileMatchesSnapshot(
      './dist/nested/json-page.html',
    );
  });
});
