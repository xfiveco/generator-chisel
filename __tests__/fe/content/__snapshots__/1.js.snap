// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FE Content Nested pages, alt templates, extra data 1`] = `
Array [
  "dist/json-page.html",
  "dist/nested/json-page.html",
  "dist/nested/more-nested/page.html",
  "dist/nested/more-nested/with-template.html",
  "dist/nested/page.html",
  "dist/scripts/app.98db252c.full.js",
  "dist/scripts/app.98db252c.js",
  "dist/scripts/app.98db252c.js.map",
  "dist/styles/main.--HASH--.css",
  "dist/styles/main.--HASH--.css.map",
  "dist/styles/main.--HASH--.full.css",
]
`;

exports[`FE Content Nested pages, alt templates, extra data 2`] = `
"<!DOCTYPE html>
<h1>Hello Hello Page</h1>
<div></div>"
`;

exports[`FE Content Nested pages, alt templates, extra data 3`] = `
"<!DOCTYPE html>
<h1>Hello Hello More</h1>
<div></div> Written on 2020-04-27T12:30:12.000Z by Jakub in ."
`;

exports[`FE Content Nested pages, alt templates, extra data 4`] = `
"<!DOCTYPE html>
<h1>Hello Hello Alt</h1>
<div>
  <p>This author writes things</p>
</div> Alternative template with post by Lubos"
`;

exports[`FE Content Nested pages, alt templates, extra data 5`] = `
"<!DOCTYPE html>
<h1>Hello </h1>
<div></div>"
`;

exports[`FE Content Nested pages, alt templates, extra data 6`] = `
"<!DOCTYPE html>
<h1>Hello JSON Title</h1>
<div></div> Alternative template with post by Jason"
`;
